### ISSUE TITLE
Serialization: `field-literal` converted to `field` since 1.39.0

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#15971]
**Describe the bug**
Regression since 1.39.0, which kinda reopens #15048, but keeping as new issue, since it seems to be different and "only" breaking all `field-literal` dimensions, which are converted to `field`.

EDIT: Could it simply be caused by Serialization correctly dumping as `field`, but the load process reads `sum` as a file reference, which then results in a `null` value on the target instance?

**To Reproduce**
See #15048 for steps.

On 1.39.0 it creates the following dump - _only showing a snippet, since #15812 caused a lot of other problems_
```
...
    filter:
    - '>'
    - - field
      - sum
      - base-type: type/Float
    - 100
...
```
On 1.38.4 it does a correct dump:
```
...
    filter:
    - '>'
    - - field-literal
      - sum
      - type/Float
    - 100
...
```

**Comment by jeff303** [comment]
Commit 8 under #15858 should fix this one.

**Comment by camsaul** [comment]
This is by design. `field-literal` isn't used anymore. Everything is a `field`

**Comment by jeff303** [comment]
> Dumping as `field` is by design. `field-literal` isn't used anymore. Everything is a `field`. So it seems like it is an issue with `load`

Yes, this is fixed by [this commit](https://github.com/metabase/metabase/pull/15858/commits/267d784913d3a9c5df81fc4b0714fdc49ebeb2f5).  The "real problem" in this case was that the string was always being interpreted by load as a fully qualified name, when in fact, it can sometimes just be a string (ex: like in this case).

**Comment by flamber** [comment]
Fixed by #15858 - will be part of 1.39.3
